yaptide/batch/batch_methods.py

```diff
@@ -259,10 +270,12 @@ def get_job_status(simulation: BatchSimulationModel, user: KeycloakUserModel, cl
         return {"job_state": EntityState.COMPLETED.value, "end_time": datetime.utcnow().isoformat(sep=" ")}
     if collect_state == "RUNNING":
         logging.debug("Collect job is in RUNNING state")
+        return {"job_state": EntityState.MERGING_RUNNING.value}
+    if job_state == "COMPLETED" and collect_state == "PENDING":
```

Comment:
what about the case when main job failed (or ran out of time) and collect job is still in queue ?
will it stay there forever ? will it be automatically removed from queue ? 

should we checked for such state  ?

Reply:
I have tried setting simulation time limit on the production. After the ARRAY_SCRIPT slurm jobs timeouts, all tasks and the main job remains in the running state.
COLLECT_SCRIPT is still executed as usual, which results in sending empty results to the backend. This incorrectly marks the simulation as completed.

Maybe we should modify the COLLECT_SCRIPT to check the states of all job tasks. If any task is not in the completed state, we then mark the main job as failed and cancel the COLLECT_SCRIPT slurm job.

Reply:
I think we should discuss that in a separate issue. In principle I'd like to collect the results if some of the jobs fail. For example user submits 500 jobs, 1 fails. Taking an average from 499 jobs is still a good way to see a reasonable output